src/runtime/Story.svelte

```diff
@@ -106,17 +142,27 @@
 </script>
 
 {#if isCurrentlyViewed}
-  {#if children}
-    {@render children(renderer.args, renderer.storyContext)}
-  {:else if template}
+  {#if template && isSnippet(template)}
     {@render template(renderer.args, renderer.storyContext)}
+  {:else if children && isSnippet(children)}
+    {#if asChild}
+      {@render children()}
+    {:else if renderer.storyContext.component}
+      <renderer.storyContext.component {...renderer.args}>
+        {@render children()}
+      </renderer.storyContext.component>
+    {:else}
+      {@render children()}
+    {/if}
+  {:else if storiesTemplate}
+    {@render storiesTemplate(renderer.args, renderer.storyContext)}
```

Comment:
does this include `children`  in the args somewhere? (not sure if dealbreaker when not but what would happen with this?

```
{#snippet template(args)}
  <h1>button</h1>
  <Button {...args}/>
{/snippet}
<Story {template}>
  Hello!
</Story>
```

would Hello! be put in the children prop of Button  in the template?

Reply:
No, it doesn't include - I'm testing it right now.

The following template will crash at runtime using `examples/Button.stories.svelte`, because it is trying to call `{@render}` on undefined `children`.

Reply:
However, if you provide `children` via meta, for example:

```js
const { Story } = defineMeta({
  args: {
    children: createRawSnippet(() => ({ render: () => "Hello!" })),
  }
});
```

Then yes, the `Hello!` will be included.

Reply:
I think in this case, when users provide both `template` and `children` snippets... we should probably throw a documented error. WDYT?

At the current moment, I managed to create an error on the TypeScript level.